end_to_end_testing/run_end_to_end.sh

```diff
@@ -14,24 +14,52 @@ set -o errtrace
 
 E2E_FAILED_TESTS=false
 E2E_FAILED_MESSAGES=()
+E2E_SKIPPED_MESSAGES=()
+
+function print_status_end_exit() {
+    if [ "${#E2E_SKIPPED_MESSAGES[@]}" -gt 0 ]; then echo -e "* * *\nSkipped Tests:"; fi
+    for skipped_test in "${E2E_SKIPPED_MESSAGES[@]}"; do
+        echo -en "- ${skipped_test}"
+    done
+
+    # Exit with a bad code if we failed any tests
+    if $E2E_FAILED_TESTS; then
+        echo -e "* * *\nFound Failed Tests"
+        for error_message in "${E2E_FAILED_MESSAGES[@]}"; do
+            echo -en "- ${error_message}"
+        done
+        exit 1
+    else
+        echo "ALL TESTS PASSED!!"
+        exit 0
+    fi
+}
+
+trap 'echo Failed unexpectedly on line ${BASH_LINENO} running ${FUNCNAME[0]}: ${BASH_COMMAND}; clean_helm; exit 1' ERR
 
 # Change to the script dir for help finding yamls
 cd "$(dirname "${0}")"
 
+# Helper to clean out all the stuff between tests
+function clean_helm() {
+    if [ -z "${E2E_LEAVE_INSTALLED_CHARTS}" ]; then
+        # Get all installed things in helm and delete/purge them
+        helm list --output json | jq '.Releases[].Name' | xargs -I {} helm delete --purge {}
```

Comment:
Do we publish a minimum helm version for Reckoner?  I was just thinking that --output json didn't show up until like 2.13.

Reply:
ah-ha - interesting. We don't actually publish and support matrix, which we could do but I think it should probably be a real effort to perform continual testing. Maybe we adhere to helm stable + previous minor version?

Reply:
Seems good to me.